authentication/MagentoAuthentication.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * This script is intended to handle Magento authentication via ZAP.
+ *
+ * Magento login form uses the anti-CSRF token called "form_key".
+ * The form key is generated once and used for the HTTP session.
+ * In order to log in we need to load the login page first to parse the form key
+ * and then send the form data along with the parsed form key.
+ *
+ * This script is based on the CasAuthentication script, so thank you guys for the example.
+ *
+ * @author Max Gopey <gopeyx@gmail.com>
+ */
+function getRequiredParamsNames(){
+    return ["loginUrl"];
+}
+
+function getOptionalParamsNames(){
+    return ["extraPostData"];
+}
+
+function getCredentialsParamsNames(){
+    return ["username", "password"];
+}
+
+function authenticate(helper, paramsValues, credentials) {
+    print("---- Magento authentication script has started ----");
+    
+    // Enable Rhino behavior, in case ZAP is running on Java 8 (which uses Nashorn)
+    if (java.lang.System.getProperty("java.version").startsWith("1.8")) {
+        load("nashorn:mozilla_compat.js");
+    }
+    
+    // Imports
+    importClass(org.parosproxy.paros.network.HttpRequestHeader);
+    importClass(org.parosproxy.paros.network.HttpHeader);
+    importClass(org.apache.commons.httpclient.URI);
+    importClass(org.apache.commons.httpclient.params.HttpClientParams);
+    importClass(java.util.regex.Pattern);
+    
+    var loginUri = new URI(paramsValues.get("loginUrl"), false);
+    
+    // Perform a GET request to the login page to get the form_key
+    var get = helper.prepareMessage();
+    get.setRequestHeader(new HttpRequestHeader(HttpRequestHeader.GET, loginUri, HttpHeader.HTTP10));
+    helper.sendAndReceive(get);
+    var formParameters = parseFormParameters(get.getResponseBody().toString());
+    
+    // Build the request body using the credentials values and the CAS values obtained from the first request
```

Comment:
Looks like this comment needs updating, guess it's a hold over from the source/template.
> CAS values....